File: LeetCode/167.two-sum-ii-input-array-is-sorted.cpp
/*
 * @lc app=leetcode id=167 lang=cpp
 *
 * [167] Two Sum II - Input Array Is Sorted
 */
#include <iostream>
#include <vector>
using namespace std;

// @lc code=start
class Solution
{
public:
    vector<int> twoSum(vector<int> &numbers, int target)
    {
        int left = 0;
        int right = numbers.size() - 1;
        vector<int> d;
        while (left <= right)
        {
            if (numbers.at(left) + numbers.at(right) == target)
            {

                d.push_back(left + 1);
                d.push_back(right + 1);
                break;
            }
            else if (numbers.at(left) + numbers.at(right) > target)
            {
                right--;
            }
            else
            {
                left++;
            }
        }

        return d;
    }
};
// @lc code=end

File: LeetCode/167.two-sum-ii-input-array-is-sorted.py
#
# @lc app=leetcode id=167 lang=python3
#
# [167] Two Sum II - Input Array Is Sorted
#

# @lc code=start
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l, r = 0, len(numbers) - 1
        while l < r:
            current_sum = numbers[l] + numbers[r]
            if current_sum == target:
                return [l + 1, r + 1]  # Adding 1 to convert to 1-indexed
            elif current_sum > target:
                r -= 1
            else:
                l += 1
        return []  # If no solution is found    
# @lc code=end

